adapter/internal/eventhub/subscription.go

```diff
@@ -129,49 +158,34 @@ func LoadSubscriptionData(configFile *config.Config) {
 		queryParamMap := make(map[string]string, 1)
 		queryParamMap[GatewayLabelParam] = configuredEnv
 		go InvokeService(ApisEndpoint, APIListMap[configuredEnv], queryParamMap, APIListChannel, 0)
-	}
-
-	go retrieveAPIListFromChannel(APIListChannel)
-	var response response
-	for i := 1; i <= len(resources); i++ {
-		response = <-responseChannel
-
-		responseType := reflect.TypeOf(response.Type).Elem()
-		newResponse := reflect.New(responseType).Interface()
-
-		if response.Error == nil && response.Payload != nil {
-			err := json.Unmarshal(response.Payload, &newResponse)
-
-			if err != nil {
-				logger.LoggerSubscription.Errorf("Error occurred while unmarshalling the response received for: "+response.Endpoint, err)
+		for {
+			data := <-APIListChannel
+			logger.LoggerSync.Debug("Receiving data for an environment")
+			if data.Payload != nil {
+				logger.LoggerSync.Debug("Payload data recieved")
+				retrieveAPIList(data)
+				break
+			} else if data.ErrorCode >= 400 && data.ErrorCode < 500 {
+				logger.LoggerSync.Errorf("Error occurred when retrieving Subscription information from the control plane: %v", data.Error)
+				health.SetControlPlaneRestAPIStatus(false)
```

Comment:
Assume a scenario where the /subscriptions endpoint returns 400 (due to some issue in Control Plane implementation) but the /apis implementation returns 200. In that case the health is going to be marked as true it seems. 
https://github.com/wso2/product-microgateway/blob/faae0b5cfe7d1427c9608980e0902892bc8cba76/adapter/internal/adapter/adapter.go#L296

Reply:
I have fixed this using another boolean to track it is set as unhealthy, and not allow a status change it it is unhealthy.